blog/2020-01-everything-github/README.md

```diff
@@ -0,0 +1,713 @@
+        on:
+          schedule:
+            - cron: 0 2 * * 1-5
+        ```
+
+   - For more information see [Events that trigger workflows](https://help.github.com/articles/events-that-trigger-workflows) and [Workflow syntax for GitHub Actions](https://help.github.com/articles/workflow-syntax-for-github-actions#on).
+
+6. Then, click on **Start commit**, add message and description if you like and click on **Commit new file**.
+
+    ![start commit](./screenshot_5_start-commit.png)
+
+4.  **Done!** 🚀.  
+   The action will run by itself the first time.
+
+Next time when you push your changes to Github, Github Actions will run the workflow we created and it will deploy your updated app on Github Pages.
+
+
+### 5.3 Automate a private repo (via `GITHUB_TOKEN`)
+
+The following example is additional,
+because we have promised for this article that you do not have to pay a cent.
+As already said, the usage of a private repos with Github Actions and Github Pages is only possible from the pro plan and higher.
+However, the big advantage is that we do not have to spend some time and effort to set up a token.
+The `GITHUB_TOKEN` is already there and we do not need to bother with it at all.
+
+Behind the scenes, Github reuses an existing feature, that is known as GitHub apps.
+In contrast to programms that authenticate with a "personal access token",
+apps do not run on behalf of a specific user - which is great.
+To make this happen, GitHub automatically installs a GitHub App on the repository and makes a corresponding "installation access token" with the name `GITHUB_TOKEN` available as a secret. 
+GitHub fetches a new token for each job, before the job begins.
+Awesome!
+
+> **ℹ️ Note**
+> 
+> As already mentioned above, the `GITHUB_TOKEN` (installation access token) will only trigger a release of a new website if the action runs in a **private repository**. In a public repo, a commit is generated, but the site does not change. See this [GitHub Community post](https://github.community/t5/GitHub-Actions/Github-action-not-triggering-gh-pages-upon-push/m-p/26869) for more info. If your repo is public, you must still use the `GH_TOKEN` (personal access token).
+
+If you have not already done so, please read the text on general workflow usage.
+The workflow looks very similar, but now we must use the other token.
+Here is a shorter example, without the tests:
+following example:
+
+```yml
+name: Deploy to GitHub Pages via angular-cli-ghpages
+
+on: [push]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 10.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10.x
+
+    - name: Prepare and deploy
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        npm install
+        npm run ng -- deploy --base-href=/everything-github-demo/ --name="<Your Git Username>" --email=your.mail@example.org
```

Comment:
Wouldn't we need `--repo` option over here?

Reply:
That's the new feature of v0.6.2 that I made for this article. The repo is now always discovered from the working directory, so that the token can be always applied! 🚀 

https://github.com/angular-schule/angular-cli-ghpages/commit/c5f7b36c437843f2472a04f1816e331f4fa7d0ff